src/parameterized_tendencies/microphysics/precipitation.jl

```diff
@@ -146,122 +145,122 @@ end
-        ᶜS_q_sno_melt = similar(Y.c, FT),
-        ᶜS_q_sno_sub_dep = similar(Y.c, FT),
-        ᶜq_rai = similar(Y.c, FT),
-        ᶜq_sno = similar(Y.c, FT),
-    )
-end
-
-"""
-Computes the rain and snow advection (down) tendency
-"""
-function precipitation_advection_tendency!(
-    Yₜ,
-    Y,
-    p,
-    colidx,
-    ::Microphysics1Moment,
-)
-    FT = Spaces.undertype(axes(Y.c))
-    (; params) = p
-
-    ρ_c = Y.c.ρ[colidx]
-
-    # helper to calculate the rain velocity
-    # TODO: assuming w_gm = 0
-    # TODO: verify translation
-
-    ρq_rai = Y.c.ρq_rai[colidx]
-    ρq_sno = Y.c.ρq_sno[colidx]
-
-    RB = Operators.RightBiasedC2F(; top = Operators.SetValue(FT(0)))
-    ᶜdivᵥ = Operators.DivergenceF2C(; bottom = Operators.Extrapolate())
-    wvec = Geometry.WVector
-    microphys_params = CAP.microphysics_params(params)
-    rain_type = CM.CommonTypes.RainType()
-    snow_type = CM.CommonTypes.SnowType()
-    velo_type = CM.CommonTypes.Blk1MVelType()
-
-    # TODO - some positivity limiters are needed
-
-    # TODO: need to add horizontal advection + vertical velocity of air
-
-    # TODO: use correct advection operators
-    if !isnothing(Yₜ)
-        @. Yₜ.c.ρq_rai[colidx] += ᶜdivᵥ(
-            wvec(
-                RB(
-                    ρq_rai * CM1.terminal_velocity(
-                        microphys_params,
-                        rain_type,
-                        velo_type,
-                        ρ_c,
-                        ρq_rai / ρ_c,
-                    ),
-                ),
-            ),
-        )
-
-        @. Yₜ.c.ρq_sno[colidx] += ᶜdivᵥ(
-            wvec(
-                RB(
-                    ρq_sno * CM1.terminal_velocity(
-                        microphys_params,
-                        snow_type,
-                        velo_type,
-                        ρ_c,
-                        ρq_sno / ρ_c,
-                    ),
-                ),
-            ),
-        )
-    end
-    return nothing
-end
-
-function precipitation_tendency!(
-    Yₜ,
-    Y,
-    p,
-    t,
-    colidx,
-    precip_model::Microphysics1Moment,
-)
-    (; ᶜS_ρe_tot, ᶜS_ρq_tot, ᶜS_ρq_rai, ᶜS_ρq_sno) = p
-    compute_precipitation_cache!(
-        Y,
-        p,
-        colidx,
-        precip_model,
-        p.atmos.turbconv_model,
-    )
-
-    if !isnothing(Yₜ)
-        @. Yₜ.c.ρ[colidx] += ᶜS_ρq_tot[colidx]
-        @. Yₜ.c.ρq_tot[colidx] += ᶜS_ρq_tot[colidx]
-        @. Yₜ.c.ρq_rai[colidx] += ᶜS_ρq_rai[colidx]
-        @. Yₜ.c.ρq_sno[colidx] += ᶜS_ρq_sno[colidx]
-
-        if :ρe_tot in propertynames(Y.c)
-            @. Yₜ.c.ρe_tot[colidx] += ᶜS_ρe_tot[colidx]
-        else
-            error(
-                "1-moment microphysics can only be coupled to ρe_tot energy variable",
-            )
-        end
-    end
-
-    precipitation_advection_tendency!(Yₜ, Y, p, colidx, precip_model)
-    return nothing
-end
+#function precipitation_cache(Y, precip_model::Microphysics1Moment)
```

Comment:
I don't think we use it anywhere. I would prefer to first update things properly for the 0-moment microphysics and the code up the 1-moment microphysics